From: Sebastien Guiriec <s-guiriec@ti.com>
Date: Wed, 19 Jan 2011 21:50:37 +0000 (+0100)
Subject: ASoC: ABE DSP: Add latency constrain on PM in order to avoid OFF mode
X-Git-Url: http://git.omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=0ada3e0f617cd68d1c38cde35a53e591372db921;hp=4ac96fd641a9ce84133bd996c5c1c38ad1fca98f

ASoC: ABE DSP: Add latency constrain on PM in order to avoid OFF mode

OFF mode can be reached by CPU idle path. Update ABE driver in
order to avoid some transistions when audio activity is running.

Change-Id: Ie121495d4a4012d49d1486ec147f44c913443769
Signed-off-by: Sebastien Guiriec <s-guiriec@ti.com>
---

Index: Kernel/sound/soc/omap/omap-abe-dsp.c
===================================================================
--- Kernel.orig/sound/soc/omap/omap-abe-dsp.c	2011-01-20 16:18:22.412069875 +0530
+++ Kernel/sound/soc/omap/omap-abe-dsp.c	2011-01-20 16:18:48.392695641 +0530
@@ -42,6 +42,7 @@
 #include <plat/omap_hwmod.h>
 #include <plat/omap_device.h>
 #include <plat/dma.h>
+#include <plat/omap-pm.h>
 
 #include <sound/core.h>
 #include <sound/pcm.h>
@@ -293,6 +294,7 @@
 		abe->opp = 0;
 		abe_stop_event_generator();
 		udelay(250);
+		omap_pm_set_max_dev_wakeup_lat(&pdev->dev, &pdev->dev, -1);
 		omap_device_set_rate(&pdev->dev, &pdev->dev, 0);
 	}
 }
@@ -1772,6 +1774,7 @@
 	pm_runtime_get_sync(&pdev->dev);
 
 	if (!abe->active++) {
+		omap_pm_set_max_dev_wakeup_lat(&pdev->dev, &pdev->dev, 1);
 		aess_restore_context(abe);
 		abe_wakeup();
 	}
